Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -359,20 +354,25 @@ int RunVertices(int vtx_attr_group, OpcodeDecoder::Primitive primitive, int coun
     // Doing early return for the opposite case would be cleaner
     // but triggers a false unreachable code warning in MSVC debug builds.
 
-    CheckCPConfiguration(vtx_attr_group);
+    if (g_needs_cp_xf_consistency_check) [[unlikely]]
+    {
+      CheckCPConfiguration(vtx_attr_group);
+      g_needs_cp_xf_consistency_check = false;
+    }
 
     // If the native vertex format changed, force a flush.
     if (loader->m_native_vertex_format != s_current_vtx_fmt ||
         loader->m_native_components != g_current_components) [[unlikely]]
     {
       g_vertex_manager->Flush();
+
+      s_current_vtx_fmt = loader->m_native_vertex_format;
+      g_current_components = loader->m_native_components;
+      auto& system = Core::System::GetInstance();
+      auto& vertex_shader_manager = system.GetVertexShaderManager();
+      vertex_shader_manager.SetVertexFormat(loader->m_native_components,
+                                            loader->m_native_vertex_format->GetVertexDeclaration());
     }
-    s_current_vtx_fmt = loader->m_native_vertex_format;
-    g_current_components = loader->m_native_components;
-    auto& system = Core::System::GetInstance();
-    auto& vertex_shader_manager = system.GetVertexShaderManager();
-    vertex_shader_manager.SetVertexFormat(loader->m_native_components,
-                                          loader->m_native_vertex_format->GetVertexDeclaration());
```

Comment:
All patches LGTM, but I actually wonder about why anything bit this code here makes a performance difference. Is this change here on its own responsible for most of the improvements of this PR?

Reply:
IIRC this one was the most noticable but all of the stuff I changed did show up in the VS profiler. It's entirely possible that they weren't an issue to begin with in the Linux or Mac OS build that were compiled with a more capable compiler.